### ISSUE TITLE
Errors occurred with last Flutter upgrade 3.0.0

### ISSUE THREAD

**Issue by carpentierchloe** [csdcorp/speech_to_text#317]
I am getting those errors on the plugin since I have installed the last version of flutter :

```
e: /Users/-/.pub-cache/hosted/pub.dartlang.org/speech_to_text-5.4.3/android/src/main/kotlin/com/csdcorp/speech_to_text/SpeechToTextPlugin.kt: (81, 8): Class 'SpeechToTextPlugin' is not abstract and does not implement abstract member public abstract fun onRequestPermissionsResult(p0: Int, p1: Array<(out) String!>, p2: IntArray): Boolean defined in io.flutter.plugin.common.PluginRegistry.RequestPermissionsResultListener
e: /Users/-/.pub-cache/hosted/pub.dartlang.org/speech_to_text-5.4.3/android/src/main/kotlin/com/csdcorp/speech_to_text/SpeechToTextPlugin.kt: (627, 5): 'onRequestPermissionsResult' overrides nothing
e: /Users/-/.pub-cache/hosted/pub.dartlang.org/speech_to_text-5.4.3/android/src/main/kotlin/com/csdcorp/speech_to_text/SpeechToTextPlugin.kt: (771, 9): Class 'ChannelResultWrapper' is not abstract and does not implement abstract member public abstract fun error(p0: String, p1: String?, p2: Any?): Unit defined in io.flutter.plugin.common.MethodChannel.Result
e: /Users/-/.pub-cache/hosted/pub.dartlang.org/speech_to_text-5.4.3/android/src/main/kotlin/com/csdcorp/speech_to_text/SpeechToTextPlugin.kt: (786, 5): 'error' overrides nothing
e: /Users/-/.pub-cache/hosted/pub.dartlang.org/speech_to_text-5.4.3/android/src/main/kotlin/com/csdcorp/speech_to_text/SpeechToTextPlugin.kt: (789, 30): Type mismatch: inferred type is String? but String was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':speech_to_text:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
```

**Comment by thandal** [comment]
See https://github.com/csdcorp/speech_to_text/pull/316

**Comment by sowens-csd** [comment]
5.5.0 is now published and should resolve this issue.

**Comment by Dieiverson** [comment]
I am getting those same errors in 5.5.0...

```
e: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\speech_to_text-5.5.0\android\src\main\kotlin\com\csdcorp\speech_to_text\SpeechToTextPlugin.kt: (811, 13): Unresolved reference: run
e: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\speech_to_text-5.5.0\android\src\main\kotlin\com\csdcorp\speech_to_text\SpeechToTextPlugin.kt: (812, 24): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/Computer/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':speech_to_text:compileDebugKotlin'.
> Compilation error. See log for more details
```

**Comment by sowens-csd** [comment]
Which version of Flutter are you building with?

**Comment by sowens-csd** [comment]
I'm not seeing the same problem building locally. Are you seeing that with the example app or with your own app? It looks to me like you probably need to update the Kotlin version in your build.gradle. Check the versions used in the example app.

**Comment by Dieiverson** [comment]
I'm using my own app. I'll try update my Kotlin version.

Thanks for helping :)

**Comment by sowens-csd** [comment]
I believe this issue is resolved with 5.5.0. If anyone is still having problems please open a new issue.
